x/incentive/keeper/payout_test.go

```diff
@@ -0,0 +1,311 @@
+	_, found = suite.keeper.GetClaim(suite.ctx, suite.addrs[0], "xrp", 1)
+	suite.True(found)
+
+}
+
+func (suite *KeeperTestSuite) setupChain() {
+	tApp := app.NewTestApp()
+	ctx := tApp.NewContext(true, abci.Header{Height: 1, Time: time.Unix(100, 0)})
+	_, addrs := app.GeneratePrivKeyAddressPairs(4)
+	authGS := app.NewAuthGenState(
+		addrs,
+		[]sdk.Coins{
+			cs(c("ukava", 400)),
+			cs(c("ukava", 400)),
+			cs(c("ukava", 400)),
+			cs(c("ukava", 400)),
+		})
+	tApp.InitializeFromGenesisStates(
+		authGS,
+	)
+	supplyKeeper := tApp.GetSupplyKeeper()
+	macc := supplyKeeper.GetModuleAccount(ctx, kavadist.ModuleName)
+	err := supplyKeeper.MintCoins(ctx, macc.GetName(), cs(c("ukava", 500)))
+	suite.NoError(err)
+
+	// add a periodic vesting account
+	ak := tApp.GetAccountKeeper()
+	acc := ak.GetAccount(ctx, addrs[0])
+	bacc := auth.NewBaseAccount(acc.GetAddress(), acc.GetCoins(), acc.GetPubKey(), acc.GetAccountNumber(), acc.GetSequence())
+	periods := vesting.Periods{
+		vesting.Period{Length: int64(1), Amount: cs(c("ukava", 100))},
+		vesting.Period{Length: int64(2), Amount: cs(c("ukava", 100))},
+		vesting.Period{Length: int64(8), Amount: cs(c("ukava", 100))},
+		vesting.Period{Length: int64(5), Amount: cs(c("ukava", 100))},
+	}
+	bva, err2 := vesting.NewBaseVestingAccount(bacc, cs(c("ukava", 400)), ctx.BlockTime().Unix()+16)
+	suite.NoError(err2)
+	pva := vesting.NewPeriodicVestingAccountRaw(bva, ctx.BlockTime().Unix(), periods)
+	ak.SetAccount(ctx, pva)
+
+	// add a validator vesting account
+	acc = ak.GetAccount(ctx, addrs[2])
+	bacc = auth.NewBaseAccount(acc.GetAddress(), acc.GetCoins(), acc.GetPubKey(), acc.GetAccountNumber(), acc.GetSequence())
+	bva, err2 = vesting.NewBaseVestingAccount(bacc, cs(c("ukava", 400)), ctx.BlockTime().Unix()+16)
+	suite.NoError(err2)
+	vva := validatorvesting.NewValidatorVestingAccountRaw(bva, ctx.BlockTime().Unix(), periods, sdk.ConsAddress{}, nil, 90)
+	ak.SetAccount(ctx, vva)
+	suite.app = tApp
+	suite.keeper = tApp.GetIncentiveKeeper()
+	suite.ctx = ctx
+	suite.addrs = addrs
+}
+
+func (suite *KeeperTestSuite) setupClaims() {
+	suite.setupChain()
+	cp1 := types.NewClaimPeriod("bnb", 1, suite.ctx.BlockTime().Add(time.Hour*168), time.Hour*8766)
+	suite.keeper.SetClaimPeriod(suite.ctx, cp1)
+	c1 := types.NewClaim(suite.addrs[0], c("ukava", 100), "bnb", 1)
+	c2 := types.NewClaim(suite.addrs[0], c("ukava", 100), "xrp", 1)
+	c3 := types.NewClaim(suite.addrs[1], c("ukava", 100), "bnb", 1)
+	suite.keeper.SetClaim(suite.ctx, c1)
+	suite.keeper.SetClaim(suite.ctx, c2)
+	suite.keeper.SetClaim(suite.ctx, c3)
+}
+
+func (suite *KeeperTestSuite) getAccount(addr sdk.AccAddress) exported.Account {
+	ak := suite.app.GetAccountKeeper()
+	return ak.GetAccount(suite.ctx, addr)
+}
+
+func (suite *KeeperTestSuite) getModuleAccount(name string) supplyExported.ModuleAccountI {
+	sk := suite.app.GetSupplyKeeper()
+	return sk.GetModuleAccount(suite.ctx, name)
+}
+
+func (suite *KeeperTestSuite) setupExpiredClaims() {
```

Comment:
Since the BeginBlocker is used to expire the valid claims in `TestDeleteExpiredClaimPeriods()` I believe we can remove this function and just use `setupClaims()`.